packages/contract-notifier/src/ContractNotifier.js

```diff
@@ -0,0 +1,119 @@
+// This notifier is used to monitor a API endpoint serving known contract addresses and their expirations.
+
+const { createEtherscanLinkMarkdown } = require("@uma/common");
+const { Datastore } = require("@google-cloud/datastore");
+const datastore = new Datastore();
+
+class ContractNotifier {
+  /**
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {String} apiEndpoint API endpoint to monitor.
+   * @param {Integer} maxTimeTillExpiration Period in seconds to look for upcoming contract expirations.
+   */
+  constructor({ logger, networker, getTime, apiEndpoint, maxTimeTillExpiration }) {
+    this.logger = logger;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.apiEndpoint = apiEndpoint;
+    this.maxTimeTillExpiration = maxTimeTillExpiration;
+  }
+
+  async checkUpcomingExpirations() {
+    this.logger.debug({
+      at: "ExpirationsNotifier",
+      message: "Checking for upcoming expirations",
+      apiEndpoint: this.apiEndpoint,
+      maxTimeTillExpiration: this.maxTimeTillExpiration,
+    });
+
+    const currentTime = await this.getTime();
+    const notifiedExpirations = await this.getNotifiedExpirations();
+    const expirationPeriod =
+      this.maxTimeTillExpiration > 259200
+        ? parseInt(this.maxTimeTillExpiration / 86400) + " days"
+        : parseInt(this.maxTimeTillExpiration / 3600) + " hours";
+    const apiUrl = this.apiEndpoint + "/global/listActive";
+    const activeContracts = await this.networker.getJson(apiUrl, { method: "post" });
+    const expiringContracts = activeContracts
```

Comment:
this code is good. can you try putting a few line breaks in-between logical chunks and throwing in a few comments to just help with readability and maintainability?

Reply:
will do

Reply:
Should I also add line breaks inside the map/filter blocks for `activeContracts`?